### DISCUSSION TITLE
Query Arguments

### DISCUSSION THREAD

**Discussion by ghost** [shurcooL/graphql#102] [Q&A]
Hello,

I am trying to replicate this query:
`query ($ids: [ArticleFilter]) {
  articleCollection(
    where: {
      OR: $ids
    }
  ) {
    items {
      slug
    }
  }
}`

Query Variables:
`[
    { "sys": { "id": "123" } },
    { "sys": { "id": "456" } },
    { "sys": { "id": "789" } }
]   `

I am unsure how to get the query type of ArticleFilter required by the endpoint I am using.

**Comment by dmitshur** [comment]
To get the "ArticleFilter" type in the variable when using this Go package, you can specify it by using a named Go type.

If your query struct looks like this (roughly):

```Go
var query struct {
	ArticleCollection struct {
		Items []struct{
			Slug string
		}
	} `graphql:"articleCollection(where: {OR: $ids})"`
}
```

Then you can define a `ArticleFilter` named type and use it in the variables map like this:

```Go
type ArticleFilter map[string]map[string]string
variables := map[string]interface{}{
	"ids": []ArticleFilter{
		{"sys": map[string]string{"id": "123"}},
		{"sys": map[string]string{"id": "456"}},
		{"sys": map[string]string{"id": "789"}},
	},
}
```

Since the variable "ids" has type `[]ArticleFilter` (slice of `ArticleFilter`), it should map to the GraphQL "[ArticleFilter]" type (list of ArticleFilter).
